workers/task-driver/src/tasks/update_wallet.rs

```diff
@@ -429,4 +453,120 @@ impl UpdateWalletTask {
             Ok(0)
         }
     }
+
+    /// Construct the appropriate event for the given wallet update type
+    /// and emit it to the event manager
+    async fn emit_event(&self) -> Result<(), UpdateWalletTaskError> {
+        let event = match &self.update_type {
+            WalletUpdateType::Deposit { .. } | WalletUpdateType::Withdraw { .. } => {
+                self.construct_external_transfer_event()?
+            },
+            WalletUpdateType::PlaceOrder { order, id, matching_pool } => {
+                self.construct_order_placement_or_update_event(order, id, matching_pool)
+            },
+            WalletUpdateType::CancelOrder { order } => {
+                self.construct_order_cancellation_event(order).await?
+            },
+        };
+
+        self.event_queue.send(event).map_err(err_str!(UpdateWalletTaskError::SendEvent))
+    }
+
+    /// Construct an external transfer event
+    fn construct_external_transfer_event(&self) -> Result<RelayerEvent, UpdateWalletTaskError> {
+        let event_id = Uuid::new_v4();
+        let event_timestamp = SystemTime::now();
+        let wallet_id = self.new_wallet.wallet_id;
+        let transfer = self
+            .transfer
+            .clone()
+            .map(|t| t.external_transfer)
+            .ok_or(UpdateWalletTaskError::Missing(ERR_MISSING_TRANSFER.to_string()))?;
+
+        Ok(RelayerEvent::ExternalTransfer(ExternalTransferEvent {
+            event_id,
+            event_timestamp,
+            wallet_id,
+            transfer,
+        }))
+    }
+
+    /// Construct either an order placement or an order update event,
+    /// depending on whether the order already exists in the new wallet
+    fn construct_order_placement_or_update_event(
+        &self,
+        order: &Order,
+        order_id: &OrderIdentifier,
```

Comment:
nit: `order_id` should come before order as it more clearly identifies the parameters